llvm/lib/CodeGen/SelectionDAG/LegalizeTypesGeneric.cpp

```diff
@@ -376,6 +376,15 @@ SDValue DAGTypeLegalizer::ExpandOp_BUILD_VECTOR(SDNode *N) {
   assert(OldVT == VecVT.getVectorElementType() &&
          "BUILD_VECTOR operand type doesn't match vector element type!");
 
+  if (VecVT.isInteger() && TLI.isOperationLegal(ISD::SPLAT_VECTOR, VecVT) &&
```

Comment:
Why do we need the SPLAT_VECTOR legality?  Isn't SPLAT_VECTOR_PARTS enough?

Reply:
I think we don't default SPLAT_VECTOR_PARTS to Expand so it started crashing other targets.